service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/jobs/JobClient.java

```diff
@@ -25,37 +32,122 @@
      *
      * @param connectionString The iot hub connection string
      * @return The instance of JobClient
-     * @throws IOException if the object creation failed
+     * @throws IOException This exception is never thrown.
      * @throws IllegalArgumentException if the provided connectionString is {@code null} or empty
+     * @deprecated because this method declares a thrown IOException even though it never throws an IOException. Users
+     * are recommended to use {@link #JobClient(String)} instead
+     * since it does not declare this exception even though it constructs the same JobClient.
      */
-    public static JobClient createFromConnectionString(String connectionString) throws IOException, IllegalArgumentException
+    @Deprecated
+    public static JobClient createFromConnectionString(String connectionString)
+        throws IOException, IllegalArgumentException
     {
-        if (connectionString == null || connectionString.length() == 0)
-        {
-            /* Codes_SRS_JOBCLIENT_21_001: [The constructor shall throw IllegalArgumentException if the input string is null or empty.] */
-            throw new IllegalArgumentException("Connection string cannot be null or empty");
-        }
+        return new JobClient(connectionString);
+    }
+
+    /**
+     * Constructor to create instance from connection string
+     *
+     * @param connectionString The iot hub connection string
+     * @return The instance of JobClient
+     */
+    public JobClient(String connectionString)
+    {
+        this(connectionString, JobClientOptions.builder()
+            .httpConnectTimeout(JobClientOptions.DEFAULT_HTTP_CONNECT_TIMEOUT_MS)
+            .httpReadTimeout(JobClientOptions.DEFAULT_HTTP_READ_TIMEOUT_MS)
+            .build());
+    }
+
+    /**
+     * Constructor to create instance from connection string
+     *
+     * @param connectionString The iot hub connection string
+     * @param options The connection options to use when connecting to the service.
+     * @return The instance of JobClient
+     */
+    public JobClient(String connectionString, JobClientOptions options)
+    {
+        this.iotHubConnectionString = IotHubConnectionStringBuilder.createIotHubConnectionString(connectionString);
+        this.hostName = this.iotHubConnectionString.getHostName();
+        this.options = options;
+    }
 
-        /* Codes_SRS_JOBCLIENT_21_003: [The constructor shall create a new DeviceMethod instance and return it.] */
-        JobClient jobClient = new JobClient();
+    /**
+     * Create a new JobClient instance.
+     *
+     * @param hostName The hostname of your IoT Hub instance (For instance, "your-iot-hub.azure-devices.net")
+     * @param credential The custom {@link TokenCredential} that will provide authentication tokens to
+     *                                    this library when they are needed. The provided tokens must be Json Web Tokens.
```

Comment:
is this normal to have so much space before the next line text begins?

Reply:
It is not, I'll fix it